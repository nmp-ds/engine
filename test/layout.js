import { setup } from "./_helpers.js";
import { describe, expect, test } from "vitest";

setup();

describe("position", () => {
  test("check static, fixed, absolute, relative and sticky values", async (t) => {
    const classes = [
      "overflow-auto",
      "overflow-hidden",
      "overflow-visible",
      "overflow-scroll",
      "overflow-x-auto",
      "overflow-x-hidden",
      "overflow-x-visible",
      "overflow-x-scroll",
      "overflow-y-auto",
      "overflow-y-hidden",
      "overflow-y-visible",
      "overflow-y-scroll",
    ];

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot(`
      "/* layer: default */
      .overflow-auto{overflow:auto;}
      .overflow-hidden{overflow:hidden;}
      .overflow-scroll{overflow:scroll;}
      .overflow-visible{overflow:visible;}
      .overflow-x-auto{overflow-x:auto;}
      .overflow-x-hidden{overflow-x:hidden;}
      .overflow-x-scroll{overflow-x:scroll;}
      .overflow-x-visible{overflow-x:visible;}
      .overflow-y-auto{overflow-y:auto;}
      .overflow-y-hidden{overflow-y:hidden;}
      .overflow-y-scroll{overflow-y:scroll;}
      .overflow-y-visible{overflow-y:visible;}"
    `);
  });
});
