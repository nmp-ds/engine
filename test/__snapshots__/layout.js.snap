// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`columns > allows values 1 to 12 1`] = `
"/* layer: default */
.columns-1{columns:1;}
.columns-10{columns:10;}
.columns-11{columns:11;}
.columns-12{columns:12;}
.columns-2{columns:2;}
.columns-3{columns:3;}
.columns-4{columns:4;}
.columns-5{columns:5;}
.columns-6{columns:6;}
.columns-7{columns:7;}
.columns-8{columns:8;}
.columns-9{columns:9;}"
`;

exports[`columns > arbitrary values 1`] = `
"/* layer: default */
.columns-\\[13\\]{columns:13;}
.columns-\\[25\\]{columns:25;}
.columns-\\[55\\]{columns:55;}"
`;

exports[`columns > columns-auto 1`] = `
"/* layer: default */
.columns-auto{columns:auto;}"
`;
